doc/examples/uuid.rst

```diff
@@ -0,0 +1,472 @@
+  the different byte-order used by the C# driver for representing UUIDs as
+  BSON Binary. The following query, on the other hand, will successfully find
+  this document::
+
+    result = collection.find_one({'_id': UUID('33221100-5544-7766-8899-aabbccddeeff')})
+
+This example demonstrates how the differing byte-order used by different
+drivers can hamper interoperability. To workaround this problem, users should
+configure their ``MongoClient`` with the appropriate
+:class:`~bson.binary.UuidRepresentation` (in this case, ``client`` in application
+``M`` can be configured to use the
+:data:`~bson.binary.UuidRepresentation.CSHARP_LEGACY` representation to
+avoid the unintuitive behavior) as described in
+:ref:`configuring-uuid-representation`.
+
+Scenario 2: Round-Tripping UUIDs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this scenario, we see how using a misconfigured
+:class:`~bson.binary.UuidRepresentation` can cause an application
+to inadvertently change the :class:`~bson.binary.Binary` subtype of a UUID
+field when round-tripping documents containing UUIDs.
+
+Consider the following situation::
+
+  from bson.codec_options import CodecOptions, DEFAULT_CODEC_OPTIONS
+  from bson.binary import Binary, UuidRepresentation
+  from uuid import uuid4
+
+  # Using UuidRepresentation.PYTHON_LEGACY stores a Binary subtype-3 UUID
+  python_opts = CodecOptions(uuid_representation=UuidRepresentation.PYTHON_LEGACY)
+  input_uuid = uuid4()
+  collection = client.testdb.get_collection('test', codec_options=python_opts)
+  collection.insert_one({'_id': 'foo', 'uuid': input_uuid})
+  assert collection.find_one({'uuid': Binary(input_uuid.bytes, 3)})['_id'] == 'foo'
+
+  # Retrieving this document using UuidRepresentation.STANDARD returns a native UUID
+  std_opts = CodecOptions(uuid_representation=UuidRepresentation.STANDARD)
+  std_collection = client.testdb.get_collection('test', codec_options=std_opts)
+  doc = std_collection.find_one({'_id': 'foo'})
+  assert doc['uuid'] == input_uuid
+
+  # Round-tripping the retrieved document silently changes the Binary subtype to 4
+  std_collection.replace_one({'_id': 'foo'}, doc)
+  assert collection.find_one({'uuid': Binary(input_uuid.bytes, 3)}) is None
+  round_tripped_doc = collection.find_one({'uuid': Binary(input_uuid.bytes, 4)})
+  assert doc == round_tripped_doc
+
+
+In this example, round-tripping the document using the incorrect
+:class:`~bson.binary.UuidRepresentation` (``STANDARD`` instead of
+``PYTHON_LEGACY``) changes the :class:`~bson.binary.Binary` subtype as a
+side-effect. **Note that this can also happen when the situation is reversed -
+i.e. when the original document is written using ``STANDARD`` representation
+and then round-tripped using the ``PYTHON_LEGACY`` representation. Note also
+that replacing ``PYTHON_LEGACY`` by ``JAVA_LEGACY`` or ``CSHARP_LEGACY`` in
+the above example produces the same behavior.**
+
+.. note:: This will not be an issue in PyMongo>=4 as starting in that version,
```

Comment:
"Starting in PyMongo 4.0 this issue will be resolved"

Reply:
done.